resqpy/grid_surface.py

```diff
@@ -1667,6 +1667,168 @@ def bisector_from_faces(grid_extent_kji: Tuple[int, int, int], k_faces: np.ndarr
     return a, is_curtain
 
 
+def first_true(array: np.ndarray) -> Optional[int]:  # type: ignore
+    """Returns the index + 1 of the first True value in the array."""
+    for idx, val in np.ndenumerate(array):
+        if val:
+            return idx[0] + 1
+
+
+def get_boundary(  # type: ignore
+        k_faces: np.ndarray, j_faces: np.ndarray, i_faces: np.ndarray, grid_extent_kji: Tuple[int, int,
+                                                                                              int]) -> Dict[str, int]:
+    """Cretaes a dictionary of the indices that bound the surface (where the faces are True).
+    
+    Args:
+        k_faces (np.ndarray): a boolean array of which faces represent the surface in the k dimension.
+        j_faces (np.ndarray): a boolean array of which faces represent the surface in the j dimension.
+        i_faces (np.ndarray): a boolean array of which faces represent the surface in the i dimension.
+        grid_extent_kji (Tuple[int]): the shape of the grid.
+
+    Returns:
+        boundary (Dict[str, int]): a dictionary of the indices that bound the surface.
+    """
+    boundary = {
+        "k_min": None,
+        "k_max": None,
+        "j_min": None,
+        "j_max": None,
+        "i_min": None,
+        "i_max": None,
+    }
+
+    indices = ['k', 'j', 'i']
+    values = [0, 1, 2]
+    for ind, faces in enumerate([k_faces, j_faces, i_faces]):
+        true_values = list(np.where(faces))
+        val = values.copy()
+        del val[ind]
+
+        ind_1 = indices[val[0]]
+        ind_2 = indices[val[1]]
+        dim_1 = true_values[val[0]]
+        dim_2 = true_values[val[1]]
+
+        if dim_1.size > 0 and dim_2.size > 0:
+            for ind, dim in zip([ind_1, ind_2], [dim_1, dim_2]):  # type: ignore
+                if boundary[f"{ind}_min"] is None:
+                    boundary[f"{ind}_min"] = min(dim)
+                elif min(dim) < boundary[f"{ind}_min"]:
+                    boundary[f"{ind}_min"] = min(dim)
+                if boundary[f"{ind}_max"] is None:
+                    boundary[f"{ind}_max"] = max(dim)
+                elif min(dim) > boundary[f"{ind}_max"]:
```

Comment:
surely this should be:
elif max(dim) > ....

Reply:
Yeah good spot